pandas/core/indexes/datetimelike.py

```diff
@@ -165,7 +168,7 @@ def __contains__(self, key: Any) -> bool:
         hash(key)
         try:
             self.get_loc(key)
-        except (KeyError, TypeError, ValueError):
+        except (KeyError, TypeError, ValueError, InvalidIndexError):
```

Comment:
what cases get here?

Reply:
Probably this one
https://github.com/pandas-dev/pandas/pull/53743/files#diff-c34a28314fc8cb12f0d2aa710f1c15f06cdfe3e48f03e658f01f99a43d4f5d09R3793-R3797